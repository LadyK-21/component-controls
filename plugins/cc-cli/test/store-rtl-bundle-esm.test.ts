import path from 'path';
import { runTests } from './run-store-tests';

runTests({
  format: 'esm',
  renderer: 'rtl',
  bundle: path.resolve(__dirname, 'bundle/component-controls.js'),
});
