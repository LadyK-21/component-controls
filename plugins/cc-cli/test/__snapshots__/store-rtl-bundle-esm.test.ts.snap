// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rtl bundle esm 1`] = `
"import * as path from 'path';
import { run } from 'axe-core';
import { reactRunDOM } from '@component-controls/test-renderers';
import '@component-controls/jest-axe-matcher';
import { loadStore } from '@component-controls/store';
import { render, act } from '@testing-library/react';
import { renderErr } from '@component-controls/test-renderers';

describe('component-controls generated', () => {
  const store = loadStore(
    require(path.resolve(__dirname, './bundle/component-controls.js')),
  );
  const renderFn = store.config.renderFn;
  Object.keys(store.docs).forEach(docId => {
    const doc = store.docs[docId];
    if (doc.stories && doc.stories.length) {
      const stories = doc.stories;
      describe(doc.title, () => {
        stories.forEach(storyId => {
          const story = store.stories[storyId];
          describe(story.name, () => {
            let rendered;
            act(() => {
              rendered = renderFn(story, doc);
            });
            if (!rendered) {
              renderErr();
              return;
            }
            it('snapshot', () => {
              const { asFragment } = render(rendered);
              expect(asFragment()).toMatchSnapshot();
            });
            it('accessibility', async () => {
              const axeResults = await reactRunDOM(rendered, run);
              expect(axeResults).toHaveNoAxeViolations();
            });
          });
        });
      });
    }
  });
});
"
`;
